enhancements/network/baremetal-ipi-network-configuration.md

```diff
@@ -0,0 +1,187 @@
+---
+title: baremetal-ipi-network-configuration
+authors:
+  - "@cybertron"
+  - "@hardys"
+reviewers:
+  - "@kirankt"
+  - "@dtantsur"
+  - "@zaneb"
+approvers:
+  - "TBD"
+creation-date: 2021-05-21
+last-updated: 2021-06-28
+status: provisional|implementable
+
+see-also:
+  - "/enhancements/host-network-configuration.md"
+  - "/enhancements/machine-config/mco-network-configuration.md"
+  - "/enhancements/machine-config/rhcos/static-networking-enhancements.md"
+
+# Baremetal IPI Network Configuration
+
+Describe user-facing API for day-1 network customizations in the IPI workflow,
+with particular focus on baremetal where such configuration is a common
+requirement.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently in the IPI flow, there is no way to provide day-1 network configuration
+which is a common requirement, particularly for baremetal users.  We can build
+on the [UPI static networking enhancements](rhcos/static-networking-enhancements.md)
+to enable such configuration in the IPI flow.
+
+## Motivation
+
+Since the introduction of baremetal IPI, a very common user request is how
+to configure day-1 networking, and in particular the following cases which are not currently possible:
+
+* Deploy with OpenShift Machine network on a tagged (non-default) VLAN
+* Deploy with OpenShift Machine network using static IPs (no DHCP)
+
```

Comment:
Deploy with OpenShift Machine network using bonded NICs (and often together with tagged VLANs) is a common user request at the same level of importance. I know it's mentioned below and if having to choose, having static IPs would be a higher prio but static/bonds/VLANs are the basic most commonly required configurations.

We also need options for setting MTUs and for bonding policies.

Reply:
Okay, these two points were the primary motivations for this work, but I'll try to capture some of the other things we'd also like to configure as well.